<html>
<body>
<h1>Pre-Processing</h1>
<hr/>
<p> This tool will read in a Scaffold <em>Samples Report</em> file (tab-delimited txt file) or a MaxQuant <em>peptides.txt</em> file and process them to generate a <em> Bait File, Prey File,</em> and <em> Inter File</em> for SAINTexpress analysis.</p>

<h4> 1) MaxQuant or Scaffold </h4>
<p>APOSTL is able to recognize either a Scaffold <em>Samples Report</em> file (tab-delimited txt file) or the <em>peptides.txt</em> file output in the MaxQuant <em>txt</em> output folder. No modifications should be made to these files. Please designate which one is being provided.</p>

<h4> 2) Scaffold or MaxQuant File Input </h4>
<p>Select the corresponding <em>Samples Report</em> or 
<em>peptides.txt</em> here.</p>

<h4> 3) Create Prey File </h4>
<p>Select whether or not you would like APOSTL to generate a prey file.</p>

<p>When making a prey file, APOSTL queries a user provided FASTA database (see below) in order to extract protein amino acid lengths and gene names. This may take several minutes depending on your computer and if your Galaxy distribution is cluster enabled. Some users may want to run SAINTexpress using the same data set while changing which baits are considered test or control. It is useful to toggle <strong>Create Prey File</strong> off in order to save time by circumventing this step as the same prey file can be used for both SAINTexpress runs.</p>

<h4> 4) Provide Uniprot FASTA database </h4>
<p> Please specify a FASTA file containing the protein sequence. You can download the latest release of Uniprot's reviewed sequences (Swiss-Prot) <a href="http://www.uniprot.org/downloads" target="_blank"> here</a>. For computing efficiency, please limit your FASTA file to your species of interest. It is generally advised to use the same FASTA database that was used for your database searches during peptide and protein ID.</p>

<h4> 5) Are You Providing Your Own Bait File? </h4>
<p>Users have two options for providing a <em>Bait</em> file. If you would like to create your own <em> Bait </em> file, select <kbd> Yes </kbd> and designate the file below. If you would like to use the <strong> Bait Create </strong> tool to create a <em> Bait </em> file within Galaxy, please select <kbd> No </kbd> and proceed to the <strong> Bait Create </strong> tool. </p>

<h4> 6) Bait File </h4>
<p> If you selected <kbd> Yes </kbd> above, please specify the <em> Bait </em> file you would like to provide. If you selected <kbd> No </kbd> above, ignore this argument and proceed to the <strong> Bait Create </strong> tool.</p>
<blockquote>
<strong>Note:</strong> Individual bait names must match to the sample names within your Scaffold or MaxQuant output exactly and must contain no whitespace characters (e.g. spaces) or dashes.
</blockquote>


<h4> 7) Bait Create </h4>
<p>Using the <strong>Bait Create</strong> tool, you can create your <em>bait.txt</em> file.
<blockquote>
<strong>Note:</strong> Individual bait names must match to the sample names within your Scaffold or MaxQuant output exactly and must contain no whitespace characters (e.g. spaces) or dashes.
</blockquote>
<p>APOSTL uses this bait file to find the user's baits of interest within a <em>Samples Report</em> or 
<em>peptides.txt</em> file and when preparing the <em> inter.txt </em> file.</p>
<hr/>
<p> Once your parameters have been finalized, please press the <kbd> Execute </kbd> button to start processing. This may take a few minutes. Once your process has been completed, you will see your <em> Bait File, Prey File,</em> and <em> Inter File</em> on the right hand side of your panel highlighted in green. </p>